include/nuttx/net/net.h

```diff
@@ -270,6 +270,10 @@ struct socket
   /* Callback instance for TCP send() or UDP sendto() */
 
   FAR struct devif_callback_s *s_sndcb;
+
+#ifdef CONFIG_DEBUG_ASSERTIONS
+  int s_sndcb_alloc_cnt;     /* The callback allocation counter */
```

Comment:
add the assertion to 
https://github.com/apache/incubator-nuttx/blob/master/net/devif/devif_callback.c#L354-L359